### PR TITLE
langs extended for dist

### PR DESCRIPTION
**Pull request by howell**

### CODE REVIEW

File: chapter/5/langs-extended-for-dist.md [hunk 1]
```diff
@@ -1,10 +1,271 @@
...
+
+One can argue that one reason the Linda model failed to catch on in iis that the primary implementations were extensions to C and Fortran. While general-purpose, C and Fortran suffer a paucity of options for creating abstractions.
+
+That being said, there are advantages to clean slate languages. A fresh start grants the designer full control over every part of the language. For some areas, such as novel type systems, this is almost a necessity. However, we have seen designs like ML5 that elegantly integrate with a base language's type system. Finally, a language designer can put their new ideas center stage, easily allowing people to see what is new and cool; in time those ideas may spread to other languages.
+
+Erik Meijer points out in {% cite salesman --file langs-extended-for-dist %} that programmers don't jump to new techonologies to access new features. Rather, in time those features make their way to old technologies. Examples abound, such as the arrival of λ in Java and of course all of Meijer's work in C# and Visual Basic.
```
Comment (pgrosu) [inline chapter/5/langs-extended-for-dist.md]: s/techonologies/technologies

File: chapter/5/langs-extended-for-dist.md [hunk 2]
```diff
@@ -1,10 +1,271 @@
...
+Many libraries can
+be described in the same way. The first-order values provided by a
+library are the primitives while the provided functions, classes, or objects perform
+combination and composition. A library can implement the primitives
+and operations such that they are in correspondence with the forms and semantics defined by a language.
+
+A theme in the literature is presenting an idea as a language and
+implementing it as a library or extension to an existing language.
+Doing so allows the authors to analyze a minimal presentation of their
+idea, but enjoy the benefits of the library/extension appraoch. Both Linda and Lasp take this approach, for example.
```
Comment (pgrosu) [inline chapter/5/langs-extended-for-dist.md]: s/appraoch/approach

File: chapter/5/langs-extended-for-dist.md [hunk 3]
```diff
@@ -1,10 +1,271 @@
...
+version of Concurrent Euclid to which the Eden team had added
+support for remote object invocation” {% cite bhjl07 --file langs-extended-for-dist %}. However, Eden
+did not take full advantage of the overlap (namely, objects) between
+the computation and distribution models.
+
+See Chapter 4 for a more extensive overview of the languages using an object-based model for distribution, such as Emerald, Argus, and E.
+
+### Batch Processing
+Another common extension is in the domain of batch processing
+large-scale data. MapReduce (and correspondingly Hadoop) is the landmark example of a programming model for distributed batch processing. Essentially, batch processing systems present a restricted programming model. For example, MapReduce programs must fit into the rigid model of two-step produce then aggregrate. A restricted model alllows the system to make more guarantees, such as for fault tolerance. Subsequently language designers have sought to increase the expressiveness of the programming models and to boost performance. Chapter 8 covers batch processing languages in more detail.
```
Comment (pgrosu) [inline chapter/5/langs-extended-for-dist.md]: s/aggregrate/aggregate

File: chapter/5/langs-extended-for-dist.md [hunk 4]
```diff
@@ -1,10 +1,271 @@
...
+actor on another node.
+
+In addition to supporting the classic actor style which places few if
+any constraints on what messages may be sent, Haskell and Scala based
+implementations leverage the host language's powerful type system to
+create more structured communication patterns. Both CloudHaskell and
+Akka, the successor to Scala Actors, provide *typed
+  channels* between actors. For example, the type
+checker will reject a program where an actor expecting to receive
+numbers is instead sent a string. Anecodatally, typed actors in Akka are not commonly used. This might suggest that errors due to incorrect message types are not that serious of a concern for users.
```
Comment (pgrosu) [inline chapter/5/langs-extended-for-dist.md]: s/Anecodatally/Anecdotally

File: chapter/5/langs-extended-for-dist.md [hunk 5]
```diff
@@ -1,10 +1,271 @@
...
+Another option is to use another language as a starting point and modify that language’s implementation for your own purposes. Think of it as forking a compiler and/or runtime and then making your own modifications. An advantage of this approach is the savings in implementation effort. If the computation models coincide then the language designer only has to implement the communication model and fault-tolerance strategy. Another plus is that users of the existing language may be more likely to consider trying or adopting the new language since it is only a step away from what they already know. A downside for maintaining any fork is keeping up to date with upstream changes.
+
+### Library
+The final approach to consider is similar to extending a language, but this time doing so only by writing code in that language to create new abstractions implementing a language model. This strategy offers similar savings in implementation effort to extending a language (perhaps moreso). But a major benefit is that is significantly easier for users to use and adopt; the only new concepts they must learn are the specifics of the programming model, as opposed to other concerns such as syntax.
```
Comment (pgrosu) [inline chapter/5/langs-extended-for-dist.md]: s/moreso/more so

**Review by pgrosu — CHANGES_REQUESTED** [review]
Really awesome Sam (especially the Macros :)!  Just a few spelling thingies.

**Comment by heathermiller** [comment]
Merging so it's up on the site, I'll follow up with comments/edits soon :)

**Pull request merged**